docs/api-reference/map.md

```diff
@@ -26,96 +26,99 @@ function App() {
 Imperative methods are accessible via a [React ref](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) or the [useMap](./use-map.md) hook.
 
 
-```js
+```tsx
 import * as React from 'react';
 import Map from 'react-map-gl';
+import mapboxgl from 'mapbox-gl';
 
 function App() {
-  const mapRef = React.useRef();
+  const mapRef = React.useRef<MapRef<mapboxgl.Map>>();
 
   const onMapLoad = React.useCallback(() => {
     mapRef.current.on('move', () => {
       // do something
     });
   }, []);
 
-  return <Map mapLib={import('mapbox-gl')} ref={mapRef} onLoad={onMapLoad} />;
+  return <Map mapLib={mapboxgl} ref={mapRef} onLoad={onMapLoad} />;
 }
 ```
 
 The [MapRef](./types.md#mapref) object exposes [Map methods](https://docs.mapbox.com/mapbox-gl-js/api/map/#map-instance-members) that **are safe to call without breaking the React bindings**. For example, `setStyle()` is hidden from the ref object, because the style is supposed to be changed by updating the `mapStyle` prop. Calling the method directly may cause the the React prop to mismatch with the underlying state, and lead to unexpected behaviors.
 
 You can still access the hidden members via `getMap()`:
 
-#### `getMap()`: MapboxMap {#getmap}
+#### `getMap()`: MapboxMap {#getmap}
 
 Returns the native [Map](https://docs.mapbox.com/mapbox-gl-js/api/map/) instance associated with this component.
 
 
 ## Properties
 
+Aside from the props listed below, the `Map` component supports all parameters of the [Map](https://docs.mapbox.com/mapbox-gl-js/api/map/#map-parameters) constructor. Beware that this is not an exhaustive list of all props. Different base map libraries may offer different options and default values. When in doubt, refer to your base map library's documentation.
+
 ### Layout options
 
-#### `id`: string {#id}
+#### `id`: string {#id}

```

Comment:
I'm seeing a bunch of edits like this that don't have a visible change.. just checking if its fine?

Reply:
The id field is auto generated